### DISCUSSION TITLE
Segmentation fault when running with MPI on cluster

### DISCUSSION THREAD

**Discussion by 0mician** [iqtree/iqtree2#199] [Q&A]
Hi everyone, super grateful to the developers of IQ-TREE. I have been using it for years and I appreciate it a lot. 

I have recently been meaning to distribute the construction of trees on multiple cluster nodes as my dataset is currently pretty large and I have access to a decent cluster:

``` bash
Alignment has 27811 sequences with 480851 columns, 183624 distinct patterns
108371 parsimony-informative, 54551 singleton sites, 317929 constant sites"""
```

I compiled IQ-TREE with MPI support, and it seems to be working for a mock dataset I have that is quite smaller (47 sequences): 
``` bash
Alignment has 52 sequences with 3845412 columns, 55532 distinct patterns
132316 parsimony-informative, 205234 singleton sites, 3507862 constant sites"""
```
However, it fails with segmentation faults on the larger alignment

``` bash
ERROR: IERROR: S--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 1 with PID 1224925 on node _redacted_ exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

These are my specs for the job:

``` bash
#SBATCH --job-name=iqtree_mpi
#SBATCH --output=iqtree_mpi_output.log
#SBATCH --error=iqtree_mpi_error.log
#SBATCH -n 4
#SBATCH -c 48
#SBATCH -N 4
#SBATCH --mem=375G
#SBATCH --time=02:00:00
#SBATCH --partition=short

module load OpenMPI/4.1.5-GCC-12.3.0

mpirun -np 4 /home/x/software/IQ-TREE/build/iqtree-mpi -s cga.aln.clipkit -m GTR+F+I+G4 -nt AUTO -bb 1000 -mem 375G
``` 

Does anyone have any ideas what I could try?

Thank you!
0mician

PS: time and number or nodes would need to be increased when I am running the actual computation of the tree, but these are the values I am using for testing that everything starts correctly.

**Comment by bqminh** [comment]
Omician, I think your alignment with >27K sequences and 400K sites is just way too large for IQ-TREE.

Can you try other software like FastTree 2 or the distance-based [DecentTree](https://github.com/iqtree/decenttree)?
